src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,97 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final double COEFFICIENT_OF_EXPANSION = 1.5;
+    private int size;
+    private T[] values;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_SIZE];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == values.length) {
+            grow();
+        }
+        values[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds: " + index);
+        }
+        if (size == values.length) {
+            grow();
+        }
+        System.arraycopy(values, index, values, index + 1, size - index);
+        values[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexValidation(index);
+        return values[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexValidation(index);
+        values[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexValidation(index);
+        T removedElement = values[index];
+        System.arraycopy(values, index + 1, values, index, size - index - 1);
+        values[--size] = null;
```

Comment:
You anyway checking about size, and user can't get this `value` after u change `size`
```suggestion
        size--;
```

Reply:
Yes, you right, i didn`t notice that, thanks.